src/BenchmarkDotNet/Toolchains/CsProj/CsProjClassicNetToolchain.cs

```diff
@@ -33,21 +36,34 @@ private CsProjClassicNetToolchain(string targetFrameworkMoniker, string name, st
         public static IToolchain From(string targetFrameworkMoniker, string packagesPath = null)
             => new CsProjClassicNetToolchain(targetFrameworkMoniker, targetFrameworkMoniker, packagesPath);
 
-        public override bool IsSupported(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver)
+        public override IEnumerable<ValidationError> Validate(BenchmarkCase benchmarkCase, ILogger logger, IResolver resolver)
         {
-            if (!base.IsSupported(benchmarkCase, logger, resolver))
-                return false;
+            var baseValidations = base.Validate(benchmarkCase, logger, resolver);
+
+            if (baseValidations.Any())
+            {
+                return baseValidations;
+            }
 
             if (!RuntimeInformation.IsWindows())
             {
-                logger.WriteLineError($"Classic .NET toolchain is supported only for Windows, benchmark '{benchmarkCase.DisplayInfo}' will not be executed");
-                return false;
+                var validationError = new ValidationError(true, $"Classic .NET toolchain is supported only for Windows, benchmark '{benchmarkCase.DisplayInfo}' will not be executed", benchmarkCase);
+
+                logger.WriteLineValidationError(validationError);
+
+                return validationError.AsEnumerable();
             }
 
             if (InvalidCliPath(customDotNetCliPath: null, benchmarkCase, logger))
-                return false;
+            {
+                var validationError = new ValidationError(true, $"InvalidCliPath, benchmark '{benchmarkCase.DisplayInfo}' will not be executed", benchmarkCase);
```

Comment:
Do not use fast exit, collect all errors and return them.

It's better to display as much as possible errors.
If the user is not on windows AND the cli path is invalid, it displays only the first message.

Reply:
I am not sure if we want to check all the errors, or just fail fast here, anyway I kep the existing behavior, not sure if we want to change that.

Reply:
You can open the validators folder, all of them may display multiple errors.

Reply:
> You can open the validators folder, all of them may display multiple errors.

I mean toolchains errors, currently it only prints one error and exits as you can see in the previous code for "IsSupported" method
I am ok to change it as well, but I am trying to be objective here, and change only the necessary for what is described in the issue, and try not have side effects
Maybe @adamsitnik can provide his opinion here.

Reply:
I like the idea of returning all errors instead the first one. 👍 

Currently if there are two or more errors:

1. The users tries to run the benchmarks, gets a single error and fixes it.
2. The users tries to run the benchmarks, gets another error and fixes it.
3. The user actually runs the benchmarks.

If we provide all errors at once we can reduce the number of steps needed and hence improve the UX.